### ISSUE TITLE
[Question] Is IgnorePkg a viable workaround for safe overlay upgrades?

### ISSUE THREAD

**Issue by erisdev** [ValShaped/rwfus#29]
As per [the wiki](https://wiki.archlinux.org/title/Pacman#Skip_package_from_being_upgraded), pacman.cong has `IgnorePkg` to prevent upgrading certain packages. If one were to add all the Steam OS default installed packages to this list, would it be safe to run `pacman -Su` on the overlay, or is there some other issue that I'm not aware of? Obviously, I'm hesitant to simply fuck around and find out.

**Comment by ValShaped** [comment]
That could very well work, yeah! Good idea!

**Comment by ValShaped** [comment]
I looked into it further, and setting IgnorePkg would require touching /etc/pacman.conf. As /etc/ is itself an overlay, that would copy it into /var/lib/overlay/etc/upper, and when the Deck updates, that file will desync from the one in the base filesystem.

We can't remount /etc/ without crashing everything, either, since that's where all the config files are.

The only other options are to wrap pacman, or create a custom package manager interface based on libalpm, which I'm diametrically apposed to doing (since that would defeat the purpose of using overlays in the first place -- and at that point you might as well just add ~/.local/some_dir to $PATH, and use your custom pacman config to install packages there.)

If you know of a solution that doesn't require reimplementing or wrapping pacman, I'm all up for it, otherwise I don't know what to do.

**Comment by erisdev** [comment]
Would it be possible to be a little rude and say that rwfus "owns" /etc/pacman.conf and re-inject a freshly generated `IgnorePkg` into a clean copy of the file before each mount? Or maybe only when it detects that the OS image has been updated, if that's possible?

**Comment by ValShaped** [comment]
The problem there is getting at the original file without unmounting /etc. Writing directly into an overlayfs upper dir while it's mounted is undefined behavior, so we can't just delete the modified file at runtime (though I've never had it fail in my testing)

**Comment by erisdev** [comment]
Gotcha. the /etc overlay doesn't belong to rwfus, then? That makes sense.

Yeah, I'm not sure what else I'd have for this, but I'll let you know if I have a stroke of genius. I'm fairly familiar with Arch, but I haven't really dug too deep into the workings of Steam OS.

**Comment by ValShaped** [comment]
Yep, the `/etc` overlay is handled by SteamOS Offload (which is part of the package `steamos-customizations`, iirc)
